### DISCUSSION TITLE
Markdown Treesitter support with lazy.nvim

### DISCUSSION THREAD

**Discussion by joshmedeski** [pwntester/octo.nvim#424] [Q&A] [answered]
I cannot figure out how to get markdown syntax highlighting with treesitter working in my setup.

<img width="1532" alt="SCR-20231004-kkxe" src="https://github.com/pwntester/octo.nvim/assets/2036594/2a3ae305-6588-421b-a685-657408b10e8f">

This is my configuration for the plugin, using [lazy.nvim](https://github.com/folke/lazy.nvim):

```lua
return {
  "pwntester/octo.nvim",
  cmd = "Octo",
  dependencies = {
    "nvim-lua/plenary.nvim",
    "nvim-telescope/telescope.nvim",
    "nvim-tree/nvim-web-devicons",
    "nvim-treesitter/nvim-treesitter",
  },
  config = function()
    vim.cmd([[hi OctoEditable guibg=none]])
    vim.treesitter.language.register("markdown", "octo")
    require("octo").setup({ enable_builtin = true })
  end,
  keys = {
    { "<leader>o", "<cmd>Octo<cr>", desc = "Octo" },
  },
}
```

- I've moved `vim.treesitter.language.register("markdown", "octo")` with not success.
- I've called `:TSBUfEnable markdown` directly in a Octo buffer and nothing happens.

If anyone else has gotten markdown syntax highlighting working with lazy.nvim I'd appreciate some guidance.

I appreciate the help, thanks!

**Comment by pwntester** [comment]
Hi Josh,

Beautiful theme in there! Have you tried clearing the `OctoEditable` hlgroup with `hi OctoEditable guibg=none guifg=none`? For reference, I register the FT alias [after](https://github.com/pwntester/dotfiles/blob/master/config/nvim/lua/pwntester/plugins/treesitter.lua#L116) setting up TreeSitter

**Comment by joshmedeski** [comment, marked as answer]
I spent the whole morning trying to figure it out. It was a combination of how octo gets the highlight groups for `OctoEditable` and some values set by my color scheme, catppuchin.

I ended up having to adjust the source code on both plugins to get it working properly with minimal configuration.

- https://github.com/pwntester/octo.nvim/pull/425
- https://github.com/catppuccin/nvim/pull/589
